File: src/bindings/ruby/test/TestParameterRule.rb
#
# This file was converted from libsbml/src/sbml/test/TestParameterRule.c
# with the help of test_c2ruby.pl (manual handling required).
#
require 'test/unit'
require 'libSBML'

class TestParameterRule < Test::Unit::TestCase

  def test_ParameterRule_free_NULL
  end

  def setup
    @@pr = LibSBML::AssignmentRule.new()
    @@pr.setL1TypeCode(LibSBML::SBML_PARAMETER_RULE)
      
  end

  def test_ParameterRule_setName
    name = "cell"
    @@pr.setVariable(name)
    assert_equal name,@@pr.getVariable
    assert_equal true, @@pr.isSetVariable
    c = @@pr.getVariable
    @@pr.setVariable(c)
    assert_equal name,@@pr.getVariable
    @@pr.setVariable("")
    assert_equal false, @@pr.isSetVariable
  end

  def test_ParameterRule_createWith
    pr = LibSBML::RateRule.new("c", "v + 1")
    pr.setL1TypeCode(LibSBML::SBML_PARAMETER_RULE)
    assert_equal LibSBML::SBML_RATE_RULE, pr.getTypeCode
    assert_equal LibSBML::SBML_PARAMETER_RULE, pr.getL1TypeCode
    assert_equal nil, pr.getNotes
    assert_equal nil, pr.getAnnotation
    assert_equal "", pr.getUnits
    assert_equal  "v + 1",pr.getFormula
    assert_equal  "c",pr.getVariable
    assert_equal LibSBML::RULE_TYPE_RATE, pr.getType
    assert_equal true, pr.isSetVariable
    assert_equal false, pr.isSetUnits
  end

  def test_ParameterRule_create
    assert_equal LibSBML::SBML_ASSIGNMENT_RULE, @@pr.getTypeCode
    assert_equal LibSBML::SBML_PARAMETER_RULE, @@pr.getL1TypeCode
    assert_equal nil, @@pr.getNotes
    assert_equal nil, @@pr.getAnnotation
    assert_equal "", @@pr.getFormula
    assert_equal "", @@pr.getUnits
    assert_equal "", @@pr.getVariable
    assert_equal LibSBML::RULE_TYPE_SCALAR, @@pr.getType
    assert_equal false, @@pr.isSetVariable
    assert_equal false, @@pr.isSetUnits
  end

  def test_ParameterRule_setUnits
    units = "cell"
    @@pr.setUnits(units)
    assert_equal units,@@pr.getUnits
    assert_equal true, @@pr.isSetUnits
      @@pr.setUnits(@@pr.getUnits)
      assert_equal units,@@pr.getUnits
      @@pr.setUnits("")
      assert_equal false, @@pr.isSetUnits
  end

end

File: src/bindings/python/test/TestParameterRule.py
#
# @file    TestParameterRule.py
# @brief   ParameterRule unit tests
# @author  Akiya Jouraku (Python conversion)
# @author  Ben Bornstein 
#
# $Id$
# $Source$
#
# This test file was converted from src/sbml/test/TestParameterRule.c
# with the help of conversion sciprt (ctest_converter.pl).
#
#<!---------------------------------------------------------------------------
# This file is part of libSBML.  Please visit http://sbml.org for more
# information about SBML, and the latest version of libSBML.
#
# Copyright 2005-2008 California Institute of Technology.
# Copyright 2002-2005 California Institute of Technology and
#                     Japan Science and Technology Corporation.
# 
# This library is free software; you can redistribute it and/or modify it
# under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation.  A copy of the license agreement is provided
# in the file named "LICENSE.txt" included with this software distribution
# and also available online as http://sbml.org/software/libsbml/license.html
#--------------------------------------------------------------------------->*/
#
import sys
import unittest
import libsbml

class TestParameterRule(unittest.TestCase):

  PR = None

  def setUp(self):
    self.PR = libsbml.AssignmentRule()
    self.PR.setL1TypeCode(libsbml.SBML_PARAMETER_RULE)
    if (self.PR == None):
      pass    
    pass  

  def tearDown(self):
    self.PR = None
    pass  

  def test_ParameterRule_create(self):
    self.assert_( self.PR.getTypeCode() == libsbml.SBML_ASSIGNMENT_RULE )
    self.assert_( self.PR.getL1TypeCode() == libsbml.SBML_PARAMETER_RULE )
    self.assert_( self.PR.getNotes() == None )
    self.assert_( self.PR.getAnnotation() == None )
    self.assert_( self.PR.getFormula() == "" )
    self.assert_( self.PR.getUnits() == "" )
    self.assert_( self.PR.getVariable() == "" )
    self.assert_( self.PR.getType() == libsbml.RULE_TYPE_SCALAR )
    self.assertEqual( False, self.PR.isSetVariable() )
    self.assertEqual( False, self.PR.isSetUnits() )
    pass  

  def test_ParameterRule_createWith(self):
    pr = libsbml.RateRule("c", "v + 1")
    pr.setL1TypeCode(libsbml.SBML_PARAMETER_RULE)
    self.assert_( pr.getTypeCode() == libsbml.SBML_RATE_RULE )
    self.assert_( pr.getL1TypeCode() == libsbml.SBML_PARAMETER_RULE )
    self.assert_( pr.getNotes() == None )
    self.assert_( pr.getAnnotation() == None )
    self.assert_( pr.getUnits() == "" )
    self.assert_((  "v + 1" == pr.getFormula() ))
    self.assert_((  "c" == pr.getVariable() ))
    self.assert_( pr.getType() == libsbml.RULE_TYPE_RATE )
    self.assertEqual( True, pr.isSetVariable() )
    self.assertEqual( False, pr.isSetUnits() )
    pr = None
    pass  

  def test_ParameterRule_free_NULL(self):
    
    pass  

  def test_ParameterRule_setName(self):
    name = "cell"
    self.PR.setVariable(name)
    self.assert_(( name == self.PR.getVariable() ))
    self.assertEqual( True, self.PR.isSetVariable() )
    if (self.PR.getVariable() == name):
      pass    
    c = self.PR.getVariable()
    self.PR.setVariable(c)
    self.assert_(( name == self.PR.getVariable() ))
    self.PR.setVariable("")
    self.assertEqual( False, self.PR.isSetVariable() )
    if (self.PR.getVariable() != None):
      pass    
    pass  

  def test_ParameterRule_setUnits(self):
    units = "cell"
    self.PR.setUnits(units)
    self.assert_(( units == self.PR.getUnits() ))
    self.assertEqual( True, self.PR.isSetUnits() )
    if (self.PR.getUnits() == units):
      pass    
    self.PR.setUnits(self.PR.getUnits())
    self.assert_(( units == self.PR.getUnits() ))
    self.PR.setUnits("")
    self.assertEqual( False, self.PR.isSetUnits() )
    if (self.PR.getUnits() != None):
      pass    
    pass  

def suite():
  suite = unittest.TestSuite()
  suite.addTest(unittest.makeSuite(TestParameterRule))

  return suite

if __name__ == "__main__":
  if unittest.TextTestRunner(verbosity=1).run(suite()).wasSuccessful() :
    sys.exit(0)
  else:
    sys.exit(1)
